sphinx_needs/functions/functions.py

```diff
@@ -250,33 +250,31 @@ def resolve_variants_options(env: BuildEnvironment):
         return
 
     variants_options = env.app.config.needs_variant_options
-    if len(variants_options) == 0:
-        # If variant_options is not provided
-        default_variant_options = ["status", "tags", "links"]
-        default_variant_options.extend(env.app.config.needs_extra_options)
-        extra_links = [extra_link["option"] for extra_link in env.app.config.needs_extra_links]
-        default_variant_options.extend(extra_links)
-        variants_options.extend(set(default_variant_options))
-
-    needs: Dict = env.needs_all_needs
-    for need in needs.values():
-        # Data to use as filter context.
-        need_context: Dict = {**need}
-        need_context.update(**env.app.config.needs_filter_data)  # Add needs_filter_data to filter context
-        _sphinx_tags = env.app.builder.tags.tags  # Get sphinx tags
-        need_context.update(**_sphinx_tags)  # Add sphinx tags to filter context
-
-        for need_option in need:
-            if need_option not in variants_options:
-                # Don't apply variant handling to options which are not in variant_options.
-                continue
-            if need[need_option] not in (None, "", []):
-                if not isinstance(need[need_option], (list, set, tuple)):
-                    option_value: str = need[need_option]
-                    need[need_option] = match_variants(option_value, need_context, env.app.config.needs_variants)
-                else:
-                    option_value = need[need_option]
-                    need[need_option] = match_variants(option_value, need_context, env.app.config.needs_variants)
+    enable_variant_handling = False
+    if len(variants_options) != 0:
+        # If variant_options is provided or not empty then activate variant handling
+        enable_variant_handling = True
+
+    if enable_variant_handling:
```

Comment:
Could this be done in one line?

``if variants_options: ...``.

An empty list is always False in Python3